poom/main_menu.py

```diff
@@ -0,0 +1,267 @@
+from abc import ABC, abstractmethod
+from os import getcwd
+from pathlib import Path
+
+import pygame
+import pygame_gui
+
+# "#2f353b"
+pygame.init()
+size = width, height = 800, 600
+screen = pygame.display.set_mode(size)
+pygame.display.set_caption("Greeting")
+root = Path(getcwd())
+background = pygame.transform.scale(
+    pygame.image.load(root / "assets" / "back.png").convert_alpha(), (width, height)
+)
+running = True
+clock = pygame.time.Clock()
+manager = pygame_gui.UIManager(size, root / "assets" / "style.json")
+animation = pygame.sprite.GroupSingle()
+
+
+class Zombie(pygame.sprite.Sprite):
+    def __init__(self):
+        super().__init__(animation)
+        self.width, self.height = 111, 162
+        self.images = [
+            pygame.transform.scale(
+                pygame.image.load(
+                    root / "assets" / "zombie" / f"{i}.png"
+                ).convert_alpha(),
+                (self.width, self.height),
+            )
+            for i in range(5)
+        ]
+        self.image = self.images[0]
+        self.rect = self.image.get_rect()
+        self.rect.x = width - self.width
+        self.rect.y = height - self.height
+        self.v = 30
+        self.x = width - self.width
+        self.k_rotates = 0
+        self.rotate = False
+        self.left = True
+        self.animation_rate = 0
+
+    def update(self, dt):
+        self.x -= self.v * dt
+        self.rect.x = self.x
+        self.animation_rate += 3 * dt
+        if not 0 < self.x <= width - self.width and not self.rotate:
+            self.v = 0
+            self.rotate = True
+            self.k_rotates = self.animation_rate
+        if not self.rotate:
+            self.image = self.images[int(self.animation_rate) % 2]
+        else:
+            if self.animation_rate - self.k_rotates < 3:
+                self.image = self.images[2 + int(self.animation_rate - self.k_rotates)]
+            else:
+                for i in range(5):
+                    self.images[i] = pygame.transform.flip(self.images[i], True, False)
+                self.v = -30 if self.left else 30
+                self.x += 2 if self.left else -2
+                self.left = not self.left
+                self.k_rotates = 0
+                self.rotate = False
+
+
+class AbstractScene(ABC):
```

Comment:
`AbstractScene` should also have `update(self, dt: float) -> None` and `render(self, surface: pg.Surface) -> None` methods